edx_sga/tasks.py

```diff
@@ -0,0 +1,309 @@
+DATA_DIR = getattr(settings, "DATA_DIR", "/edx/app/edxapp/data")
+
+
+def start_downloading_submissions(course_id, block_id, location, user):
+    """
+    helper to start downloading task
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user (django.contrib.auth.models.Use): staff user
+    """
+    if course_id and block_id and location:
+        zip_student_submissions.delay(
+            course_id,
+            block_id,
+            location,
+            user.username
+        )
+
+
+def _mk_dir(path):
+    """
+    Create a new directory.
+
+    Args:
+        path (str): path of new directory
+    """
+    _rm_file(path)
+    os.mkdir(path)
+
+
+def _rm_file(path):
+    """
+    Removes file if it exist.
+
+    Args:
+        path (str): path of new directory
+    """
+    if os.path.exists(path):
+        rmtree(path)
+
+
+def get_user_by_id(user_id):
+    """
+    Returns user object
+
+    Args:
+        user_id (int): user id
+    """
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        user = None
+
+    return user
+
+
+def _get_user_name(user):
+    """
+    returns user name
+
+    Args:
+        user (django.contrib.auth.models.User): user instance
+
+    return:
+        (str): user name
+    """
+    if user:
+        user_name = user.username
+    else:
+        user_name = uuid4().hex
+
+    return user_name
+
+
+def user_name_by_id(user_id):
+    """
+    returns user name by given user id
+
+    Args:
+        user_id (int): user id
+
+    return:
+        (str): user name
+    """
+    if user_id:
+        user = get_user_by_id(user_id)
+        user_name = _get_user_name(user)
+    else:
+        user_name = uuid4().hex
+
+    return user_name
+
+
+def _get_submissions(student_id, course_id, block_id):
+    """
+    Returns submissions of student
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+    """
+    params = {
+        "student_id": student_id,
+        "course_id": course_id,
+        "item_id": block_id,
+        "item_type": 'sga',
+    }
+    return submissions_api.get_submissions(params)
+
+
+def _compress_student_submissions(block_id, course_id, location, destination_path):
+    """
+    prepare files for download
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        destination_path (str): folder from where we will zip files and facilitate for download
+    """
+    from edx_sga.sga import BLOCK_SIZE
+    submissions_students = SubmissionsStudent.objects.filter(
+        course_id=course_id,
+        item_id=block_id
+    )
+    for submissions_student in submissions_students:
+        submissions = _get_submissions(submissions_student.student_id, course_id, block_id)
```

Comment:
In order to prepare the submission zip file, we should just be iterating through all student submissions for this particular block, correct? Unless I'm missing something, we should _definitely_ be using [get_all_submissions](https://github.com/edx/edx-submissions/blob/master/submissions/api.py#L423) instead of this approach where we get students first, then their most recent submission. We should also get rid of `_get_submissions`, which is redundant

Reply:
i dont see edx-submission app install in edx platform, how can i access it, show i install it on sga?

Reply:
[Right here in our edx-platform fork](https://github.com/mitodl/edx-platform/blob/mitx/ginkgo/requirements/edx/github.txt#L78). We're already using the submissions app in SGA (see [here](https://github.com/mitodl/edx-sga/pull/187/files#diff-52fbfd5003bb036038e11d03721c4253R27)). If the submissions app wasn't installed in edx-platform, then this import and SGA in general would not work. This practice of depending on edx-platform to satisfy that dependency, instead of declaring that dependency in this project, is bad practice, but that's what we have.